airflow/utils.py

```diff
@@ -617,11 +619,20 @@ def handle_timeout(self, signum, frame):
         raise AirflowTaskTimeout(self.error_message)
 
     def __enter__(self):
-        signal.signal(signal.SIGALRM, self.handle_timeout)
-        signal.alarm(self.seconds)
+        try:
+            signal.signal(signal.SIGALRM, self.handle_timeout)
+            signal.alarm(self.seconds)
+        except Exception as e:
```

Comment:
I am worried about the code in the try/except throwing an exception when it is not being run via LocalExecutor. Maybe we can check if we are running in a local executor and maybe catching a narrower scope of exceptions.